simulation/halsim_gui/src/main/native/cpp/Mechanism2D.cpp

```diff
@@ -0,0 +1,319 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "Mechanism2D.h"
+
+#include <cmath>
+
+#include <hal/SimDevice.h>
+#include <imgui.h>
+
+#define IMGUI_DEFINE_MATH_OPERATORS
+
+#include <imgui_internal.h>
+#include <wpi/json.h>
+#include <wpi/raw_istream.h>
+#include <wpi/raw_ostream.h>
+#include "HALSimGui.h"
+#include <string>
+#include <wpi/math>
+#include <hal/simulation/SimDeviceData.h>
+#include "portable-file-dialogs.h"
+
+using namespace halsimgui;
+
+static HAL_SimDeviceHandle devHandle = 0;
+static wpi::StringMap<ImColor> colorLookUpTable;
+static std::unique_ptr<pfd::open_file> m_fileOpener;
+static std::string previousJsonLocation = "Not empty";
+
+struct BodyConfig {
```

Comment:
This and DrawLineStruct and Mechanism2DInfo should be in an anonymous namespace.